subsys/net/lib/app/net_app.c

```diff
@@ -2066,6 +2066,9 @@ int _net_app_ssl_mainloop(struct net_app_ctx *ctx)
 			if (ctx->tls.mbedtls.ssl_ctx.hdr) {
 				/* Needed to skip the protocol header */
 				hdr_len = ctx->tls.mbedtls.ssl_ctx.hdr->len;
+				/* set packet family for checks below */
+				net_pkt_set_family(pkt,
+					net_context_get_family(net_ctx));
```

Comment:
Shouldn't this setup be part of  subsys/net/ip/net_pkt.c instead? It got removed from there with commit #753daa6, but only because it wasn't needed by those changes. At the time this usage wasn't caught, so perhaps there may be more code with the same semantics.

Now that commit #753daa6 and later have correctly computed pkt->data_len, couldn't that information be used here instead of re-computing the data length? Or is this only about IP header lengths and not net_pkt payload length?

Reply:
@pfl You are probably correct.  Along those lines, I can re-implement code in net_pkt.c net_pkt_get() to be executed regardless of whether slab != &rx_pkts to do the following:
- net_pkt_set_context
- net_pkt_set_iface
- net_pkt_set_family

As these could probably be used elsewhere in the code regardless of the slab used.

To clarify: the checks are used for header lengths, not payload data length, so the data_len calc doesn't help.

Reply:
If you can get by with getting the address family for the checks below (under `#ifdefs`), why not just call `net_context_get_family(net_ctx)` instead or `net_pkt_family(pkt)`?  Otherwise, yeah, @pfl's suggestion makes sense.

Reply:
@lpereira I can get by with the using net_context_get_family in the checks below, however, I think @pfl was wondering if other code may have been affected by the commit removing the initialization of several packet values based on the slap used.

Reply:
@pfl can you let me know if my understanding of your previous statements is correct above?

Reply:
Yes it is. But now that I look again on the commit/source code, it does set all the details, but only when slab != &rx_pkts. There was a reason for this, but @jukkar is away right now so I can't re-iterate on the why... So you end up using the rx_pkts slab in your code?

Reply:
[net_core.c](https://github.com/rveerama1/zephyr/blob/master/subsys/net/ip/net_core.c#L108) will start setting (family, proto) on incoming(rx) packets. Then UDP handler handover the packet to net_context connection ([packet_received](https://github.com/rveerama1/zephyr/blob/master/subsys/net/ip/net_context.c#L2230)). But in [net_app with tls](https://github.com/rveerama1/zephyr/blob/master/subsys/net/lib/app/net_app.c#L2056) creates **new** packet for decrypted payload. This is a fresh new packet which doesn't have any information (iface, family, context. etc..) because of slab != &rx_pkts. @pfl MTU handling required for only Tx packets and all setting calls slipped to if condition which might be the cause.
I would suggest net_pkt_set_context, net_pkt_set_iface, net_pkt_set_family, net_pkt_set_ip_hdr_len could be set irrespective of "slab != &rx_pkts" condition.

Reply:
> I would suggest net_pkt_set_context, net_pkt_set_iface, net_pkt_set_family, net_pkt_set_ip_hdr_len could be set irrespective of "slab != &rx_pkts" condition.

That seems to be the change we are looking for.

Reply:
Today, I noticed that I saw getting a hidden error in net_pkt:
[net/conn] [DBG] net_conn_input: (0x00403500): [0] match found cb 0x0000ed83 ud 0x00400660 rank 0x33
*[net/pkt] [ERR] net_frag_read: Not enough data to read*
*[net/udp] [ERR] net_udp_get_hdr: {assert: 'frag' failed}*
[lib/lwm2m_engine] [DBG] lwm2m_udp_receive: checking for reply from [2001:db8::2]

This doesn't seem to affect the incoming data when I use this patch as the registration works, but when I re-add the default packet setup in net_pkt.c it doesn't work at all.

Still taking a look at this.